### DISCUSSION TITLE
求助，特定域名的dns解析，添加了规则的情况下也坚持不走代理

### DISCUSSION THREAD

**Discussion by Starseeker-H** [vernesong/OpenClash#4161] [Q&A]
出现问题的域名是 raw.githubusercontent.com，可能是校园网防火墙的问题，校园网内不走代理通过任意dns都无法解析到这个域名的ip，只要不在校园网就可以解析，直接装在设备上的 clash for xxx 也可以使校园网内这个域名正常访问，检查可以发现使用了规则 ```DomainKeyword (github)```

现在的问题是，openclash中添加规则 ```- DOMAIN-KEYWORD,github,自由选择```，且放在最上面：
```
rules:
- DST-PORT,7895,REJECT
- DST-PORT,7892,REJECT
- IP-CIDR,198.18.0.1/16,REJECT,no-resolve
- DOMAIN-KEYWORD,github,自由选择
- DST-PORT,7895,REJECT
- DST-PORT,7892,REJECT
```
希望可以让连接到路由器的设备在校园网内正常访问这个域名，结果却是github的其他url都能正常访问，检查连接发现规则也确实生效了，但只有这个域名下的访问结果和直连校园网一样，无法获取ip。

百思不得其解，虽然直接不连路由器用流量就可以，代理都不用开，但这到底是为什么呢？

op运行于windows的vm虚拟机下，使用mate内核，ua3f，配置文件与备份文件（已移除订阅链接）：
[mine.txt](https://github.com/user-attachments/files/17654911/mine.txt)
[Backup-OpenClash-Exclude-Cores-2024-11-07-09-49-25.zip](https://github.com/user-attachments/files/17654968/Backup-OpenClash-Exclude-Cores-2024-11-07-09-49-25.zip)

**Comment by vernesong** [comment]
redir-host是这样的，你可以使用nameserver-policy指定dns，并且respect-rules：true让dns根据规则走代理

**Reply by Starseeker-H** [reply]
respect-rules：true 看meta的文档说是 “dns 连接遵守路由规则”？但设置了没效果。

经过测试发现阿里的DNS没受这个影响设了
```
 nameserver-policy:
  'raw.githubusercontent.com': '223.5.5.5'
```
但这个也没用，最后在第二DNS服务器里加了223.5.5.5，对raw.githubusercontent.com，才有用……

**Reply by vernesong** [reply]
查看debug日志有DNS返回记录，如果污染你就指定一个国外的DNS
